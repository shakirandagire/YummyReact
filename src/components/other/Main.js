import React, { Component } from 'react';
import { Link } from 'react-router-dom';

/**
 * Component for rendering the main page
 */
class Main extends Component {
  render() {
    return (
      <div className="my_container backgimg">
        <nav className="navbar navbar-inverse">
          <div className="container-fluid">
            <div className="navbar-header">
              <a className="navbar-brand title"><Link to="/">Yummy Recipe App</Link></a>

            </div>
          </div>
        </nav>
        <div className="mycontent">
          <h3> Do you love cooking and eating amazing food </h3>
          <h3> And cannot keep track of your favorite recipe</h3>
          <h3> We have got you </h3>
          <h3> View our amazing recipes by simply login</h3>
          <button type="submit" className="btn btn-default formsave login"><Link to="/login"> Login </Link></button>
          <h4>Do not have an account with us:</h4>
          <button type="submit" className="btn btn-default formsave register"><Link to="/register"> Sign Up </Link></button>
        </div>
      </div>

    );
  }
}
export default Main;
