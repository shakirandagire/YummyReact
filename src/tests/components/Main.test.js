import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Main from '../../components/other/Main';

const component = shallow(<Main />);

describe('<Main/>', () => {
  it('should render properly', () => {
    expect(component.length).toBe(1);
  });

  it('should render main', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('should login on click', () => {
    component.find('.btn-default .login').simulate('click');
  });
  it('should register on click', () => {
    component.find('.btn-default .register').simulate('click');
  });
});
