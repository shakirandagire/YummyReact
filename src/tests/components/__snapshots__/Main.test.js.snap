// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render main 1`] = `
<div
  className="my_container backgimg"
>
  <nav
    className="navbar navbar-inverse"
  >
    <div
      className="container-fluid"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand title"
        >
          <Link
            replace={false}
            to="/"
          >
            Yummy Recipe App
          </Link>
        </a>
      </div>
    </div>
  </nav>
  <div
    className="mycontent"
  >
    <h3>
       Do you love cooking and eating amazing food 
    </h3>
    <h3>
       And cannot keep track of your favorite recipe
    </h3>
    <h3>
       We have got you 
    </h3>
    <h3>
       View our amazing recipes by simply login
    </h3>
    <button
      className="btn btn-default formsave"
      type="submit"
    >
      <Link
        replace={false}
        to="/login"
      >
         Login 
      </Link>
    </button>
    <h4>
      Do not have an account with us:
    </h4>
    <button
      className="btn btn-default formsave"
      type="submit"
    >
      <Link
        replace={false}
        to="/register"
      >
         Sign Up 
      </Link>
    </button>
  </div>
</div>
`;
